pkg/i2gw/providers/nginx/common/filters/factory.go

```diff
@@ -0,0 +1,340 @@
+package filters
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/nginx/common"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// Header defines an HTTP Header.
+type Header struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+// FilterType represents the type of filter to create
+type FilterType string
+
+const (
+	RequestHeaderModifierFilter  FilterType = "RequestHeaderModifier"
+	ResponseHeaderModifierFilter FilterType = "ResponseHeaderModifier"
+	URLRewriteFilter             FilterType = "URLRewrite"
+	RequestRedirectFilter        FilterType = "RequestRedirect"
+)
+
+// HeaderModifierOptions contains options for header modification filters
+type HeaderModifierOptions struct {
+	// Set headers as key-value pairs
+	SetHeaders map[string]string
+	// Headers to remove (list of header names)
+	RemoveHeaders []string
+	// For NGINX CRD compatibility - explicit header structs
+	SetHeadersStruct    []Header
+	RemoveHeadersStruct []string
+}
+
+// URLRewriteOptions contains options for URL rewrite filters
+type URLRewriteOptions struct {
+	Path string
+}
+
+// RequestRedirectOptions contains options for request redirect filters
+type RequestRedirectOptions struct {
+	// HTTP status code for the redirect
+	StatusCode int
```

Comment:
nit: for style, the comment should start with the name of the variable

```suggestion
	// StatusCode represents the HTTP status code for the redirect
	StatusCode int
```